app/serializers/assignment_serializer.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+class AssignmentSerializer < ActiveModel::Serializer
+  attributes :id, :mission_id, :role, :new_assignment, :name, :destroy
+
+  def new_assignment
+    false
+  end
```

Comment:
This and `destroy` seem a bit hacky. Can you design the component code to work without them, just defaulting to false if they're not there?